requests/_internal_utils.py

```diff
@@ -25,3 +25,18 @@ def to_native_string(string, encoding='ascii'):
             out = string.decode(encoding)
 
     return out
+
+
+def unicode_is_ascii(u_string):
+    """Determine if unicode string only contains ASCII characters.
+
+    :param str u_string: unicode string to check. Must be unicode
+        and not Python 2 `str`.
+    :rtype: bool
+    """
+    assert isinstance(u_string, str)
+    try:
+        u_string.encode('ascii')
+        return True
+    except UnicodeError:
```

Comment:
It should only be possible to see a `UnicodeEncodeError` here now, as we're sure that the instance is `str`. Let's tighten the exception handling so we don't accidentally mask weird bugs.